internal/controller/template_controller.go

```diff
@@ -301,23 +318,114 @@ func (r *TemplateReconciler) getHelmChartFromChartRef(ctx context.Context, chart
 	return helmChart, nil
 }
 
+func (r *ClusterTemplateReconciler) validateCompatibilityAttrs(ctx context.Context, template *hmc.ClusterTemplate) error {
+	management := new(hmc.Management)
+	if err := r.Client.Get(ctx, client.ObjectKey{Name: hmc.ManagementName}, management); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil // nothing to update in template's status
```

Comment:
I think we should mark it as invalid with a message of something like "waiting for Management creation to complete validation". and requeue with some static interval until it's created